### PR TITLE
Bug 1678158 - backport oauth2 support to rhcc

### PR DESCRIPTION
**Pull request by jmrodri**
* fix debug statement
* add version log string
* copy the skip_verify_tls setting to the reg config
* pass client around so we can skip tls on image get

### CODE REVIEW

**Comment by jmrodri** [comment]
### Before the patch
```
...
[2019-03-19T15:29:32.623Z] [INFO] - Attempting bootstrap...
[2019-03-19T15:29:32.623Z] [INFO] - AnsibleBroker::Bootstrap
[2019-03-19T15:29:32.623Z] [DEBUG] - Dao::BatchGetRaw
[2019-03-19T15:29:32.624Z] [DEBUG] - Successfully loaded [ 0 ] objects from etcd dir [ /spec ]
[2019-03-19T15:29:32.788Z] [DEBUG] - RHCCRegistry::LoadImages
[2019-03-19T15:29:32.788Z] [DEBUG] - Using https://registry.access.stage.redhat.com/v1/search?q=%22%2A-apb%22 to source APB images
[2019-03-19T15:29:32.943Z] [ERROR] - unable to retrieve image names for registry rh - Get https://registry.access.stage.redhat.com/v1/search?q=%22%2A-apb%22: x509: certificate signed by unknown authority
[2019-03-19T15:29:32.943Z] [WARNING] - registry: 0x13f5930 was unable to complete bootstrap - Get https://registry.access.stage.redhat.com/v1/search?q=%22%2A-apb%22: x509: certificate signed by unknown authority
[2019-03-19T15:29:32.943Z] [ERROR] - Failed to bootstrap on startup!
[2019-03-19T15:29:32.943Z] [ERROR] - all registries failed on bootstrap
```

**Comment by jmrodri** [comment]
### After the patch
```
...
[2019-03-19T20:22:49.349Z] [INFO] - == REGISTRY CX == 
[2019-03-19T20:22:49.349Z] [INFO] - Name: rh
[2019-03-19T20:22:49.349Z] [INFO] - Type: rhcc
[2019-03-19T20:22:49.349Z] [INFO] - Url: https://registry.access.stage.redhat.com
[2019-03-19T20:22:49.349Z] [WARNING] - skipping verification of registry TLS certificate per adapter configuration
[2019-03-19T20:22:49.349Z] [DEBUG] - Creating filter for registry: rh
[2019-03-19T20:22:49.349Z] [DEBUG] - whitelist: [.*-apb$]
[2019-03-19T20:22:49.349Z] [DEBUG] - blacklist: []
...
[2019-03-19T20:22:49.805Z] [DEBUG] - GET /v2/ successful without token
[2019-03-19T20:22:49.805Z] [DEBUG] - RHCCRegistry::LoadImages
[2019-03-19T20:22:49.805Z] [DEBUG] - Using https://registry.access.stage.redhat.com/v1/search?q=%22%2A-apb%22 to source APB images
[2019-03-19T20:22:50.312Z] [DEBUG] - Properly unmarshalled image response
[2019-03-19T20:22:50.312Z] [DEBUG] - Filter applied against registry: rh
[2019-03-19T20:22:50.313Z] [DEBUG] - APBs passing white/blacklist filter:
[2019-03-19T20:22:50.313Z] [DEBUG] - -> rhpam-7/rhpam-apb
[2019-03-19T20:22:50.313Z] [DEBUG] - -> openshift3/mariadb-apb
[2019-03-19T20:22:50.313Z] [DEBUG] - -> codeready-workspaces-beta/server-apb
[2019-03-19T20:22:50.313Z] [DEBUG] - -> openshift3/automation-broker-apb
[2019-03-19T20:22:50.313Z] [DEBUG] - -> openshift3/mediawiki-apb
[2019-03-19T20:22:50.313Z] [DEBUG] - -> cnv-tech-preview/kubevirt-apb
[2019-03-19T20:22:50.313Z] [DEBUG] - -> rhv4-tech-preview/ovirt-flexvolume-driver-apb
[2019-03-19T20:22:50.313Z] [DEBUG] - -> cnv-tech-preview/import-vm-apb
[2019-03-19T20:22:50.313Z] [DEBUG] - -> openshift3/mysql-apb
[2019-03-19T20:22:50.313Z] [DEBUG] - -> openshift3/postgresql-apb
[2019-03-19T20:22:50.313Z] [DEBUG] - RHCCAdapter::FetchSpecs
...
[2019-03-19T20:22:50.711Z] [DEBUG] - openshift3/mariadb-apb
[2019-03-19T20:22:50.711Z] [DEBUG] - RHCCAdapter::LoadSpec
[2019-03-19T20:22:50.711Z] [DEBUG] - Registry::imageToSpec
[2019-03-19T20:22:50.991Z] [DEBUG] - adapter::imageToSpec -> Got plans [{ID: Name:dev Description:This plan deploys a single MariaDB ...
...
[2019-03-19T20:22:53.293Z] [DEBUG] - Successfully converted Image registry.access.stage.redhat.com/openshift3/postgresql-apb:v3.9.68 into Spec
[2019-03-19T20:22:53.293Z] [INFO] - Validating specs...
[2019-03-19T20:22:53.293Z] [NOTICE] - All specs passed validation!
[2019-03-19T20:22:53.344Z] [NOTICE] - Broker successfully bootstrapped on startup
[2019-03-19T20:22:53.344Z] [DEBUG] - RefreshInterval: 10m0s
[2019-03-19T20:22:53.344Z] [DEBUG] - calling NewSecureServingOptions
[2019-03-19T20:22:53.458Z] [DEBUG] - Creating k8s apiserver
[2019-03-19T20:22:53.459Z] [DEBUG] - Unable to get broker.cluster_url from config
[2019-03-19T20:22:53.46Z] [NOTICE] - Listening on https://[::]:1338
[2019-03-19T20:22:53.46Z] [NOTICE] - Ansible Service Broker Starting
```

File: pkg/registries/adapters/adapter.go [hunk 1]
```diff
@@ -48,22 +48,28 @@ const BundleSpecLabel = "com.redhat.apb.spec"
...
+	SkipVerifyTLS bool
 }
 
 // Retrieve the spec from a registry manifest request
 func imageToSpec(log *logging.Logger, req *http.Request, image string) (*apb.Spec, error) {
+	return imageToSpecWithClient(http.DefaultClient, log, req, image)
+}
+
+// Retrieve the spec from a registry manifest request
+func imageToSpecWithClient(client *http.Client, log *logging.Logger, req *http.Request, image string) (*apb.Spec, error) {
```
Comment (shawn-hurley) [inline pkg/registries/adapters/adapter.go]: 👍

**Pull request merged**